### PR TITLE
Ecf 108391510 activity planning

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/scripts/controllers/lesson.js [hunk 1]
```diff
@@ -1,119 +1,187 @@
...
-                                  el.content + '</div>';
+    var slides = selectedLesson.slides.sort(function(a, b) {
+      return a.position - b.position;
+    });
 
-          });
+    var slideCount = slides.length;
+
+    function activityReportTemplate(activity) {
+      return '<div style="height:' + docHeight +
```
Comment (ericcf) [inline app/scripts/controllers/lesson.js]: here's some technical debt...

File: app/scripts/controllers/ActivityPlanningController.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+(function() {
+  'use strict';
+
+  function ActivityPlanningController(Resources) {
+    this.savePlannedActivity = function savePlannedActivity() {
+      Resources.save(Resources.NAMES.PlannedActivities, {
+        value: this.activity,
```
Comment (eschlange) [inline app/scripts/controllers/ActivityPlanningController.js]: How does the activity get set for this method?

Comment (ericcf) [inline app/scripts/controllers/ActivityPlanningController.js]: Oops, good catch! This controller is actually not being used - I need to delete the code.

File: app/scripts/controllers/lesson.js (hunk 1 above)
Comment (eschlange) [inline app/scripts/controllers/lesson.js]: Ouch. Should these be converted to modals, or entirely separate views?

Comment (ericcf) [inline app/scripts/controllers/lesson.js]: Much of this controller logic should be refactored into a template and directive, which I will do if I have to add any new logic here.

File: test/support/array-find-polyfill.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+if (!Array.prototype.find) {
+  Array.prototype.find = function(predicate) {
+    if (this === null) {
+      throw new TypeError('Array.prototype.find called on null or undefined');
+    }
+    if (typeof predicate !== 'function') {
+      throw new TypeError('predicate must be a function');
+    }
+    var list = Object(this);
+    var length = list.length >>> 0;
```
Comment (eschlange) [inline test/support/array-find-polyfill.js]: This technique of number retrieval is new to me. Seems very useful given JavaScript's type system.

Comment (ericcf) [inline test/support/array-find-polyfill.js]: Yeah - I actually stole it from here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

**Comment by eschlange** [comment]
Many updates, looks good.

**Pull request merged**